src/common/utxobased/engine/makeUtxoEngine.ts

```diff
@@ -449,21 +449,32 @@ export async function makeUtxoEngine(
         coin: coinInfo.name
       })
 
-      const privateKeys = await Promise.all(
-        psbt.inputs.map(async ({ hash, index }) => {
-          const txid = Buffer.from(hash).reverse().toString('hex')
-
-          const [utxo] = await processor.fetchUtxos({
-            utxoIds: [`${txid}_${index}`]
-          })
-          if (utxo == null) throw new Error('Invalid UTXO')
-
-          const address = await processor.fetchAddress(utxo.scriptPubkey)
-          if (address?.path == null) throw new Error('Invalid script pubkey')
-
-          return walletTools.getPrivateKey({ path: address.path, xprivKeys })
-        })
-      )
+      // Use the privateKeys (WIFs) from spendInfo, otherwise get them from the
+      // PBST inputs.
+      const privateKeys =
```

Comment:
This should not be a ternary. Just use the normal `let/if/else` pattern.

Reply:
I really wish we had `do` expressions. I dislike `let` for reasons. Can I use a IIFE with if/else?

Reply:
> IIFE

Definitely